crates/deploy/src/lib.rs

```diff
@@ -83,39 +206,110 @@ pub async fn deploy(
 
     // set up client with auth token and compression
     let token: MetadataValue<_> = format!("Bearer {}", auth_token).parse()?;
-    let mut client = DeployerClient::new(InterceptedService::new(channel, {
-        let token = token.clone();
+    let client = DeployerClient::new(InterceptedService::new(channel, {
         move |mut req: Request<()>| {
             req.metadata_mut().insert("authorization", token.clone());
             Ok(req)
         }
     }))
     .send_compressed(CompressionEncoding::Gzip)
     .accept_compressed(CompressionEncoding::Gzip);
+    Ok(client)
+}
 
-    // iterate over all files to deploy (everything in the build directory)
-    let file_paths = WalkDir::new(path.as_ref().join("build"))
-        .into_iter()
-        .filter_map(|e| e.ok())
-        .filter(|e| e.metadata().map(|x| x.is_file()).unwrap_or(false))
-        .map(|x| x.into_path());
+enum FileRequest {
+    SendMD5(String),
```

Comment:
Same here regarding UTF-8.

I would prefer to use the appropriate type for the job, unless this is an encoded path, (in which case a newtype may be more clear)